### PR TITLE
[EC-224] Add Cosmos DB to FIMS infra

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Add resource group
Add cosmos account and related dbs and containers
Cosmos DB account is NoSql and serverless; containers' max RUs reduced to 1000
Add related workflows
Resources renaming

### Motivation and context

New FIMS infrastructure (both TF configuration and Azure resources)

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### CODE REVIEW

File: .github/workflows/pr_infra.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: (PR) Infra WestEurope
```
Comment (gunzip) [inline .github/workflows/pr_infra.yaml:1]: can we use another region?

File: .github/workflows/release_repository.yaml [hunk 2]
```diff
@@ -6,15 +6,12 @@ on:
     paths:
       - "infra/repository/**"
 
-env:
-  DIR: "infra/repository"
-
 jobs:
-  code_review:
+  release:
     uses: ./.github/workflows/call_release_infra.yaml
```
Comment (gunzip) [inline .github/workflows/release_repository.yaml:11]: infra or repository?

File: infra/prod/_modules/cosmos/cosmos_container-client.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+resource "azurerm_cosmosdb_sql_container" "client" {
+
+  name                = "client"
+  resource_group_name = var.resource_group_name
+  account_name        = module.cosmosdb_account_fims.name
+  database_name       = module.cosmosdb_database_fims.name
+
+  partition_key_path    = "/organizationId"
+  partition_key_version = 2
```
Comment (gunzip) [inline infra/prod/_modules/cosmos/cosmos_container-client.tf:9]: what's this value for?

File: .github/workflows/pr_infra.yaml (hunk 1 above)
Comment (Krusty93) [reply .github/workflows/pr_infra.yaml:1]: Not yet, but probably we need to start to think about it and eventually have a discussion about common resources like vnets, apim and app gateway
I'd keep fims in we for a while, but we need to have the discussion before going live

File: .github/workflows/release_repository.yaml (hunk 2 above)
Comment (Krusty93) [reply .github/workflows/release_repository.yaml:11]: small fix, don't know if it does worth to open another pr

File: infra/prod/_modules/cosmos/cosmos_container-client.tf (hunk 3 above)
Comment (Krusty93) [reply infra/prod/_modules/cosmos/cosmos_container-client.tf:9]: `2` stands for "large partition keys", which definition is explained [here](https://learn.microsoft.com/en-us/azure/cosmos-db/large-partition-keys?tabs=dotnetv3).
I copied the value from the current Cosmos account, assuming the development team already evaluated it.
Any thoughts? cc @lucacavallaro @michaeldisaro

Comment (lucacavallaro) [reply infra/prod/_modules/cosmos/cosmos_container-client.tf:9]: It can be removed, I think it was a typo.

Comment (lucacavallaro) [reply infra/prod/_modules/cosmos/cosmos_container-client.tf:9]: (it can be removed everywhere)

Comment (gunzip) [reply infra/prod/_modules/cosmos/cosmos_container-client.tf:9]: it looks like the docs recommend it anyway, maybe we should put this value into the module (as default)

File: .github/workflows/release_repository.yaml (hunk 2 above)
Comment (Krusty93) [reply .github/workflows/release_repository.yaml:11]: Probably it does because of some issue I found today..

**Review by gunzip — APPROVED** [review]

File: infra/prod/_modules/cosmos/cosmos_container-client.tf
Comment (lucacavallaro) [inline infra/prod/_modules/cosmos/cosmos_container-client.tf:1]: imho we should avoid mixing `snake_case` with `kebab-case`

Comment (Krusty93) [reply infra/prod/_modules/cosmos/cosmos_container-client.tf:1]: The idea was not mixing two naming convention styles but instead using `-` as a separator between the resource type and the resource name.
If you wish I could change to use snake case over all

**Pull request merged**